### DISCUSSION TITLE
Image 전송과 데이터 전송 API 분리

### DISCUSSION THREAD

**Discussion by nitronium102** [SWM-Space-Odyssey/WeddingMate_BackEnd#21] [Ideas]
하지만 "보통 이미지와 JSON데이터를 함께 보내지 않습니다. API를 전송할 때는 보통 content/type을 application/json으로 전송하는데 이것은 multipart/formdata 방식이 아니기 때문입니다." 이라고 합니다

이에 따라, 방법은 2가지 인데

**1) 분리해서 요청하기**
서버는 이미지를 저장하고, 저장된 이미지의 id(또는 파일명)를 클라이언트에 전송합니다.
클라이언트는 이미지의 id와 전송할 데이터를 application/json으로 전송합니다.

**2) base64 인코딩**
이미지를 base64로 인코딩해서 application/json으로 전송할 데이터와 base64로 인코딩된 이미지를 함께 전송합니다.
이 중 하나를 골라서 개선해야할 것 같습니다!
찾아보니 2번 방법은 디비에 base64 로 저장할경우, 차지하는 용량이 많이 늘고, 트래픽도 는다는 단점이 있다고 합니다! 이에 따라 개인적인 의견으로는 1번이 보다 적합할 것 같습니다!

원본: https://github.com/EFUB-LakkuLakku/LakkuLakku-Back/discussions/114

**Comment by lina1919** [comment]
태멘 멘토링 때 이 부분을 여쭤봤는데요.
현업에서도 이미지 요청을 분리하여 처리한다고 합니다. 다만 그 이유는 test code 때문은 아니고 이미지 용량이 클 경우 업로드에 너무 오랜 시간이 걸리는 것을 방지하기 위함이라고 합니다.
따라서 저희 서비스에서도 이미지는 유저가 업로드 하는 즉시 s3에 업로드하고, return된 이미지 url을 client에서 받아 이를 객체 생성 요청에 다시 담아서 요청하는 방식으로 구현하는 것이 좋을 것 같습니다. 
유저가 업로드는 하였지만 추후 이를 삭제하여 그 이미지가 db에 반영되지 않은 경우가 있을 수 있기 때문에 추후 s3와 db를 비교하여 db에 반영되지 않은 이미지를 삭제하는 logic을 구현하여야 합니다.
다만 s3 업로드가 객체 생성보다 선행되는 경우 s3 파일 경로에 객체 id 값을 반영할 수 없다는 점이 문제일 듯 하여 이 부분에 대해 더 논의해보면 좋을 것 같습니다.
